### ISSUE TITLE
pybind11 not installed correctly

### ISSUE THREAD

**Issue by jluttine** [NixOS/nixpkgs#53497]
## Issue description

pybind11 doesn't install correctly. pyopencl depends on pybind11 but it cannot find it even if adding it as a dependency. Perhaps this comment is related to the actual cause of the problem: https://github.com/NixOS/nixpkgs/pull/35027#issuecomment-409713656

### Steps to reproduce

In `pkgs/development/python-modules/pyopencl/default.nix`, add `pybind11` to function arguments and to `propagatedBuildInputs`. Then, try to build pyopencl:

```
$ nix-shell -p pythonPackages.pyopencl
these derivations will be built:
  /nix/store/qrk3mxx0z44wm4xqsgs96r3z9gglmv8a-python2.7-pyopencl-2018.2.1.drv
building '/nix/store/qrk3mxx0z44wm4xqsgs96r3z9gglmv8a-python2.7-pyopencl-2018.2.1.drv'...
unpacking sources
unpacking source archive /nix/store/03w05mvm2m9g6k1nsyzc6jxhdjd5xz5w-pyopencl-2018.2.1.tar.gz
source root is pyopencl-2018.2.1
setting SOURCE_DATE_EPOCH to timestamp 1540276226 of file pyopencl-2018.2.1/setup.cfg
patching sources
substituteStream(): WARNING: pattern 'pytest>=2' doesn't match anything in file 'setup.py'
configuring
building
---------------------------------------------------------------------------
Pybind11 is not installed.
---------------------------------------------------------------------------
Very likely, the build process after this message will fail.

Simply press Ctrl+C and type
python -m pip install pybind11
to fix this. If you don't, the build will continue
in a few seconds.

[1] https://pybind11.readthedocs.io/en/stable/
---------------------------------------------------------------------------
Download error on https://pypi.org/simple/pybind11/: [Errno -2] Name or service not known -- Some packages may not be found!
Couldn't find index page for 'pybind11' (maybe misspelled?)
Download error on https://pypi.org/simple/: [Errno -2] Name or service not known -- Some packages may not be found!
No local packages or working download links found for pybind11

---------------------------------------------------------------------------
Sorry, your build failed. Try rerunning configure.py with different options.
---------------------------------------------------------------------------
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 371, in <module>
    main()
  File "setup.py", line 367, in main
    zip_safe=False)
  File "/build/pyopencl-2018.2.1/aksetup_helper.py", line 21, in setup
    setup(*args, **kwargs)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/setuptools/__init__.py", line 142, in setup
    _install_setup_requires(attrs)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/setuptools/__init__.py", line 137, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/setuptools/dist.py", line 586, in fetch_build_eggs
    replace_conflicting=True,
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 780, in resolve
    replace_conflicting=replace_conflicting
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1063, in best_match
    return self.obtain(req, installer)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1075, in obtain
    return installer(requirement)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/setuptools/dist.py", line 653, in fetch_build_egg
    return cmd.easy_install(req)
  File "/nix/store/pm0xfsz6lihd7fsl1dfpvxg8assy745p-python2.7-bootstrapped-pip-18.1/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 673, in easy_install
    raise DistutilsError(msg)
distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pybind11')
builder for '/nix/store/qrk3mxx0z44wm4xqsgs96r3z9gglmv8a-python2.7-pyopencl-2018.2.1.drv' failed with exit code 1
error: build of '/nix/store/qrk3mxx0z44wm4xqsgs96r3z9gglmv8a-python2.7-pyopencl-2018.2.1.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.88, NixOS, 19.03.git.44b02b5 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - nixpkgs: `/etc/nixpkgs`
```

**Comment by jluttine** [comment]
Yes, exactly. But adding pybind11 as a build dependency doesn't solve the build issue because it's not found, probably because nix pybind11 doesn't contain the python package. So pybind11 nix derivation needs fixing, if I understand correctly.

**Comment by FlorianFranzen** [comment]
"Not installed correctly" is a bit of an overstatement: Pybind11 is a header only library, so it really only needs to be copied into the right place.

The issue stems from the fact that you can install it as a "regular" package or as a python package, which changes how the headers can be discovered and used.

Therefore I opened a pull request to add pybind11 in form of a python package in #54888.

**Comment by jluttine** [comment]
Great, thanks!

**Comment by dotlambda** [comment]
This has been fixed, hasn't it?

**Comment by FlorianFranzen** [comment]
Yes, this should be fixed with #54888.
